pkg/storegateway/series_chunks.go

```diff
@@ -11,17 +11,23 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/prometheus/model/labels"
 
+	"github.com/grafana/mimir/pkg/storage/tsdb"
 	"github.com/grafana/mimir/pkg/storegateway/storepb"
 	util_math "github.com/grafana/mimir/pkg/util/math"
 	"github.com/grafana/mimir/pkg/util/pool"
 )
 
-// Mimir compacts blocks up to 24h. Assuming a 5s scrape interval as worst case scenario,
-// and 120 samples per chunk, there could be 86400 * (1 / 5) * (1 / 120) = 144 chunks for
-// a series in the biggest block. Using a slab size of 1000 looks a good trade-off to support
-// high frequency scraping without wasting too much memory in case of queries hitting a low
-// number of chunks (across series).
-const seriesChunksSlabSize = 1000
+const (
+	// Mimir compacts blocks up to 24h. Assuming a 5s scrape interval as worst case scenario,
+	// and 120 samples per chunk, there could be 86400 * (1 / 5) * (1 / 120) = 144 chunks for
+	// a series in the biggest block. Using a slab size of 1000 looks a good trade-off to support
+	// high frequency scraping without wasting too much memory in case of queries hitting a low
+	// number of chunks (across series).
+	seriesChunksSlabSize = 1000
+
+	// Selected so that an individual chunk's data typically fits within the slab size
+	chunkBytesSlabSize = 10 * tsdb.EstimatedMaxChunkSize
```

Comment:
I'm wondering if this isn't too much. Specifically - `tsdb.EstimatedMaxChunkSize`. I digged around prometheus and I found this https://github.com/grafana/mimir/blob/19b5a3a8de779b4c96b48e17fba75c8ef442419b/vendor/github.com/prometheus/prometheus/storage/series.go#L247-L253

which was introduced with prometheus/prometheus PR 8582. This is also the code we use in the mimir compactor (@pracucci can confirm, but I'm 96% certain). It limits the number of samples per chunk to 120. 

If we take an average of 1.5 bytes per sample, a chunk is 180 bytes. With a scrape interval of 30s a series will have 24 chunks. With 5s scrapes (kind of an extreme) that's 144 chunks. 144*180 ~= 25KB. WDYT about having the slab size as 25K, so we fit the chunks for a series on the same slab?

On the other hand we have many series in a batch (5000) and those slabs could be possibly reused, so a slab size of 160K is less likely to go to waste.

Reply:
we can also implement something similar to `BucketedBytes` that works with a slab pool and worry less about tuning the slab size correctly

Reply:
I don't have a strong opinion about the optimal slab size. I would just keep it divisible by 2, or maybe a power of 2, like 32KB. Given the most common case each chunk is few hundred bytes, I think a lower value than 160000 is fine, as you suggest Dimitar.

Reply:
> we can also implement something similar to BucketedBytes that works with a slab pool and worry less about tuning the slab size correctly

I'm not super sure about it. `BucketedBytes` adds extra complexity which I wanted to avoid here. The actual purpose of this PR is to setup the foundation to rediscuss `BucketedBytes`.

Reply:
`BucketedBytes` allows to not care about the size of chunks. But I think we can afford to make some assumptions about the size of chunks.

Maybe we should also measure how effective this size is - counters for how empty is a slab is after it goes past the lookback range. Thoughts?

Reply:
The main difference is that in `BucketedBytes` slabs can be huge (MBs) while here slabs are small, so a waste may be negligible. I think we're making it overcomplicated for this use case.

Reply:
👍  then a size of 16KiB or 8KiB for a slab sounds good to me

Reply:
I opted for 16KiB now, would it make sense to run another test on dev with that size?

Reply:
I would just do a quick one and then we merge it. Thanks!